src/utils/autoSave/excludeSavedAmendments.ts

```diff
@@ -0,0 +1,52 @@
+import { isEmpty, isEqual } from "lodash"
+import { AmendmentKeys, Amendments, OffenceField, ResultQualifierCode } from "types/Amendments"
+
+const handleArray = (
+  map: Map<AmendmentKeys, (OffenceField<number> | OffenceField<string> | ResultQualifierCode)[]>,
+  amendmentField: AmendmentKeys,
+  amendments: Amendments,
+  savedAmendments: Amendments
+) => {
+  const newValues: (OffenceField<number> | OffenceField<string> | ResultQualifierCode)[] = []
+  const amendmentsArray = amendments[amendmentField] as
+    | OffenceField<number>[]
+    | OffenceField<string>[]
+    | ResultQualifierCode[]
+
+  const savedAmendmentsArray =
+    (savedAmendments[amendmentField] as OffenceField<number>[] | OffenceField<string>[] | ResultQualifierCode[]) ?? []
+
+  amendmentsArray.forEach((amendment) => {
+    const matchedSaved = savedAmendmentsArray.find(
+      (saveAmendment) => amendment.offenceIndex === saveAmendment.offenceIndex
+    )
+
+    if (!isEqual(amendment, matchedSaved)) {
+      newValues.push(amendment)
+    }
+  })
+
+  if (!isEmpty(newValues)) {
+    map.set(amendmentField, newValues)
+  }
+}
+
+const excludeSavedAmendments = (
+  amendmentFields: AmendmentKeys[],
```

Comment:
I wonder if it's necessary to pass the amendmentFields as an argument

Reply:
From the AutoSave point of view, it only wants to save a single value, e.g. `asn`

And from the `Submit Exceptions`, it removes the `offenceReasonSequence`, which gets added to the notes.